microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceDiagnosticsParticipant.java

```diff
@@ -185,7 +208,10 @@ private void validateAsynchronousAnnotation(MethodDeclaration node, List<Diagnos
 			} catch (Exception e) {
 				throw e;
 			}
-			if ((!(FUTURE_TYPE_UTILITY.equals(methodReturnTypeString)) && !(COMPLETION_STAGE_TYPE_UTILITY.equals(methodReturnTypeString)))) {
+			if ((!(FUTURE_TYPE_UTILITY.equals(methodReturnTypeString)) && 
+					!(COMPLETION_STAGE_TYPE_UTILITY.equals(methodReturnTypeString)) &&
+					!(UNI_TYPE_UTILITY.equals(methodReturnTypeString))
+					)) {
 				IOpenable openable = context.getTypeRoot().getOpenable();
 				Diagnostic d = context.createDiagnostic(context.getUri(),
 						"The annotated method does not return an object of type Future or CompletionStage",
```

Comment:
Might as well expand this to include `io.smallrye.mutiny.Uni` ? I guess we could just call it Uni for now.

Reply:
Correct me if I'm wrong, I think UNI_TYPE_UTILITY on line 213 covers `io.smallrye.mutiny.Uni`

Reply:
Yes, but the error message, right here, makes no indication of this.

Reply:
Ah good catch